File: Programmers/Lv_1/로또의최고순위와최저순위.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int rankNum(int n) {
    switch (n) {
    case 6:
        return 1;
    case 5:
        return 2;
    case 4:
        return 3;
    case 3:
        return 4;
    case 2:
        return 5;
    default:
        return 6;
    }
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int correctCnt = 0;
    int zeroCnt = 0;
    int highRank = 0, lowRank = 0;

    for (int i = 0; i < 6; i++) {
        if (find(win_nums.begin(), win_nums.end(), lottos[i]) != win_nums.end()) correctCnt++;
        if (lottos[i] == 0) zeroCnt++;
    }
    highRank = rankNum(correctCnt + zeroCnt);
    lowRank = rankNum(correctCnt);

    answer.push_back(highRank);
    answer.push_back(lowRank);
    
    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int data;
    vector<int> lottos, win_nums, result;

    for (int i = 0; i < 6; i++) {
        cin >> data;
        lottos.push_back(data);
    }

    for (int i = 0; i < 6; i++) {
        cin >> data;
        win_nums.push_back(data);
    }

    result = solution(lottos, win_nums);

    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }

    return 0;
}

File: Programmers/Lv_1/로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    rank = [6, 6, 5, 4, 3, 2, 1]
    
    cnt_0 = lottos.count(0)
    answer = 0
    for i in win_nums:
        if i in lottos:
            answer += 1
    
    return rank[cnt_0 + answer], rank[answer]


def solution2(lottos, win_nums):
    zero = lottos.count(0)
    a = [x for x in lottos if x in win_nums]
    max = zero + len(a)
    min = len(a)
    
    max = 7 - max if max >= 1 else 6
    min = 7 - min if min >= 1 else 6
    
    return [max, min]